frontend/src/components/Profile/Profile.tsx

```diff
@@ -32,122 +48,155 @@ const InnerProfile: React.FC = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
+    if (user && user.user_metadata) {
+      setFormData((prev) => ({
+        ...prev,
+        ...user.user_metadata as UserMetadata,
+      }));
+    }
+    if (user && user.email) {
+      setFormData((prev) => ({ ...prev, email: user.email as string }));
+    }
+  }, [user]);
+
+  const handleSave = async () => {
+    if (user) {
+      const { data, error } = await client.auth.updateUser({ data: formData });
+      if (error) {
+        console.error("Error updating user's data:", error);
+      }
+      if (data) {
+        console.log("User updated:", data);
+      }
+    }
+    setIsEditing(false);
```

Comment:
I think we should only `setIsEditing` to false when the save is successful right? If the save fails we should leave it in the editing state

Reply:
yeah, we should include the save fails situation.